### ISSUE TITLE
The Datapack Goes Into Safe Mode in 1.20.5 (5.1.2 pre-1)

### ISSUE THREAD

**Issue by Loweredgames** [NeunEinser/bingo#219]
![2024-03-29_23 13 07](https://github.com/NeunEinser/bingo/assets/55211569/c11ec744-c39f-4fe1-a4d4-2a72705a5dd0)

**Comment by NeunEinser** [comment]
Ah, wow 24w13a broke something.

Use 23w12a for now.

**Comment by davepusey** [comment]
Firstly, bug title says 23w13a when it should be 24w13a

Log snippet from 24w13a when it fails to load...

```
[12:24:40] [Render thread/ERROR]: Registry loading errors:
> Errors in registry minecraft:dimension_type:
>> Errors in element fetchr:default:
java.lang.IllegalStateException: Failed to parse fetchr:dimension_type/default.json from pack file/Fetchr.zip

Caused by: java.lang.IllegalStateException: No key max_inclusive in MapLike[{"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}]; No key min_inclusive in MapLike[{"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}]; Not a number: {"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}
```

**Comment by davepusey** [comment]
Confirming that adding an overlay folder for datapack version 37, with the revised syntax in `pack_format_37\data\fetchr\dimension_type\default.json` does now allow it to load successfully.

![image](https://github.com/NeunEinser/bingo/assets/787745/df922cf2-6401-47b1-b01b-d79f9cf587ea)

**Comment by Loweredgames** [comment]
> Firstly, bug title says 23w13a when it should be 24w13a
> 
> Log snippet from 24w13a when it fails to load...
> 
> ```
> [12:24:40] [Render thread/ERROR]: Registry loading errors:
> > Errors in registry minecraft:dimension_type:
> >> Errors in element fetchr:default:
> java.lang.IllegalStateException: Failed to parse fetchr:dimension_type/default.json from pack file/Fetchr.zip
> 
> Caused by: java.lang.IllegalStateException: No key max_inclusive in MapLike[{"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}]; No key min_inclusive in MapLike[{"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}]; Not a number: {"type":"minecraft:uniform","value":{"max_inclusive":7,"min_inclusive":0}}
> ```

fix name

**Comment by Loweredgames** [comment]
> Ah, wow 24w13a broke something.
> 
> Use 24w12a for now.

I noticed that there are no overlays in the repository but only in the release

**Comment by davepusey** [comment]
Yeah, Neun got a fancy script thingy that generates them all automatically.

**Comment by NeunEinser** [comment]
Yep, you can run `python neunscript.py` and it generates the release as it is published in /dist.

Basically, I can add a comment #NEUN_SCRIPT since|until <pack_format> in function files to control overlays on a per-line-basis.

Advantage is, when making changes that affect multiple pack format versions, I don't have to edit multiple files, disadvantage is, the code I run when developing is not necessarily the same as when published, and I could introduce bugs in neun script comments which are unnoticeable until actually building. #220 could be an example of it, either that or I broke it after coding the points scoreboard.

**Comment by NeunEinser** [comment]
In this case, since it is a JSON file, I would need to implicitly copy it into that folder tho, unless I extend my scripting to work fot JSON fields. I already had to do that for the structure files as well, you will notice that those actually exist in the overlay.

**Comment by Loweredgames** [comment]
> In this case, since it is a JSON file, I would need to implicitly copy it into that folder tho, unless I extend my scripting to work fot JSON fields. I already had to do that for the structure files as well, you will notice that those actually exist in the overlay.

👍
